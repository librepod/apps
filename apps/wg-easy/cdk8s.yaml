language: typescript
app: bun run main.ts
