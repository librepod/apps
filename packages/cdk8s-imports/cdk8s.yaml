language: typescript
app: bun run main.ts
imports:
  - k8s
  - argoproj.io_crds.yaml
  - traefik.io_crds.yaml
