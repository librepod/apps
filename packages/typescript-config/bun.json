{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "compilerOptions": {
    "lib": ["ESNext"],
    "target": "ESNext",
    "module": "ESNext",
    "moduleDetection": "force",
    "allowJs": false,
    "moduleResolution": "nodenext",
    "allowImportingTsExtensions": true,
    "verbatimModuleSyntax": true,
    "noEmit": true,
    "strict": true,
    "skipLibCheck": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,
    "useUnknownInCatchVariables": true,
    "noPropertyAccessFromIndexSignature": true,
    "resolveJsonModule": true,
    "esModuleInterop": true
  }
}
