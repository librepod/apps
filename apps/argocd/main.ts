import { App } from 'cdk8s'

import { ArgoProjectsChart } from './projects'
import { ArgoSystemAppsChart } from './system-apps'
import { ArgoUserAppsChart } from './user-apps'

// Projects
const argoProjects = new App()
new ArgoProjectsChart(argoProjects, 'projects', { disableResourceNameHashes: true })
argoProjects.synth()

// System Apps
const argoSystemApps = new App()
new ArgoSystemAppsChart(argoSystemApps, 'librepod-system-apps', { disableResourceNameHashes: true })
argoSystemApps.synth()

// User Apps
const argoUserApps = new App()
new ArgoUserAppsChart(argoUserApps, 'librepod-user-apps', { disableResourceNameHashes: true })
argoUserApps.synth()
