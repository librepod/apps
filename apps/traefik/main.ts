import { name as appName, version } from './package.json'
import { Construct } from 'constructs'
import { Helm, App, Chart, ChartProps } from 'cdk8s'
import * as utils from '@repo/cdk8s-utils'

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props)

    new Helm(this, appName, {
      repo: 'https://traefik.github.io/charts',
      chart: appName,
      namespace: appName,
      releaseName: appName,
      version: version,
      values: utils.parseValues('values.yaml')
    });
  }
}

const app = new App()
new MyChart(app, appName, { disableResourceNameHashes: true })
app.synth()
