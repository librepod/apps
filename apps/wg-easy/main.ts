import { name as appName } from './package.json'
import { Construct } from 'constructs'
import { App, Chart, ChartProps } from 'cdk8s'

import { createIngress, createIngressUdp } from '@repo/cdk8s-utils'
import { createDeployment } from './lib/deployment'

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props)

    createDeployment(this)
    createIngress(this, `${appName}`)
    createIngressUdp(this, `${appName}-udp`, 51820)
  }
}

const app = new App()
new MyChart(app, appName, { disableResourceNameHashes: true })
app.synth()
