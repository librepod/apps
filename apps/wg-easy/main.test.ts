import { test, expect, describe } from 'bun:test';
import { MyChart } from './main'
import { Testing } from 'cdk8s'

describe('Chart', () => {
  test('synth', () => {
    const app = Testing.app()
    const chart = new MyChart(app, 'test-chart')
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot()
  })
})
