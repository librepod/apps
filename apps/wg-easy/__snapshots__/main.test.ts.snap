// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart synth 1`] = `
[
  {
    "apiVersion": "v1",
    "data": {
      "PASSWORD": "admin",
      "WG_DEFAULT_ADDRESS": "10.6.0.x",
      "WG_DEFAULT_DNS": "192.168.2.167",
      "WG_HOST": "ee.relay.librepod.org",
      "WG_MTU": "1280",
      "WG_PORT": "6000",
    },
    "immutable": false,
    "kind": "ConfigMap",
    "metadata": {
      "name": "test-chart-cm-c87e1663",
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "name": "test-chart-deployment-c8cfd183",
    },
    "spec": {
      "minReadySeconds": 0,
      "progressDeadlineSeconds": 600,
      "replicas": 1,
      "selector": {
        "matchLabels": {
          "cdk8s.io/metadata.addr": "test-chart-Deployment-c8550d6d",
        },
      },
      "strategy": {
        "rollingUpdate": {
          "maxSurge": "25%",
          "maxUnavailable": "25%",
        },
        "type": "RollingUpdate",
      },
      "template": {
        "metadata": {
          "labels": {
            "cdk8s.io/metadata.addr": "test-chart-Deployment-c8550d6d",
          },
        },
        "spec": {
          "automountServiceAccountToken": false,
          "containers": [
            {
              "envFrom": [
                {
                  "configMapRef": {
                    "name": "test-chart-cm-c87e1663",
                  },
                },
              ],
              "image": "ghcr.io/wg-easy/wg-easy:10",
              "imagePullPolicy": "Always",
              "name": "main",
              "ports": [
                {
                  "containerPort": 51821,
                  "name": "http",
                  "protocol": "TCP",
                },
                {
                  "containerPort": 51820,
                  "name": "wg",
                  "protocol": "UDP",
                },
              ],
              "readinessProbe": {
                "failureThreshold": 3,
                "initialDelaySeconds": 5,
                "periodSeconds": 10,
                "successThreshold": 1,
                "tcpSocket": {
                  "port": 51821,
                },
                "timeoutSeconds": 1,
              },
              "resources": {
                "limits": {
                  "cpu": "1500m",
                  "memory": "2048Mi",
                },
                "requests": {
                  "cpu": "1000m",
                  "memory": "512Mi",
                },
              },
              "securityContext": {
                "allowPrivilegeEscalation": true,
                "privileged": true,
                "readOnlyRootFilesystem": false,
                "runAsNonRoot": false,
              },
              "volumeMounts": [
                {
                  "mountPath": "/etc/wireguard",
                  "name": "pvc-test-chart-pvc-c80b1b35",
                },
                {
                  "mountPath": "/dev/net/tun",
                  "name": "tun",
                },
              ],
            },
          ],
          "dnsPolicy": "ClusterFirst",
          "hostNetwork": false,
          "restartPolicy": "Always",
          "securityContext": {
            "fsGroupChangePolicy": "Always",
            "runAsNonRoot": false,
          },
          "setHostnameAsFQDN": false,
          "terminationGracePeriodSeconds": 30,
          "volumes": [
            {
              "name": "pvc-test-chart-pvc-c80b1b35",
              "persistentVolumeClaim": {
                "claimName": "test-chart-pvc-c80b1b35",
                "readOnly": false,
              },
            },
            {
              "hostPath": {
                "path": "/dev/net/tun",
                "type": "CharDevice",
              },
              "name": "tun",
            },
          ],
        },
      },
    },
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "name": "test-chart-deployment-service-c835a1d2",
    },
    "spec": {
      "externalIPs": [],
      "ports": [
        {
          "name": "wg",
          "port": 51820,
          "protocol": "UDP",
          "targetPort": 51820,
        },
        {
          "name": "http",
          "port": 80,
          "protocol": "TCP",
          "targetPort": 51821,
        },
      ],
      "selector": {
        "cdk8s.io/metadata.addr": "test-chart-Deployment-c8550d6d",
      },
      "type": "ClusterIP",
    },
  },
  {
    "apiVersion": "v1",
    "kind": "PersistentVolumeClaim",
    "metadata": {
      "name": "test-chart-pvc-c80b1b35",
    },
    "spec": {
      "accessModes": [
        "ReadWriteMany",
      ],
      "resources": {
        "requests": {
          "storage": "0.1Gi",
        },
      },
      "storageClassName": "nfs-client",
      "volumeMode": "Filesystem",
    },
  },
]
`;
